rustls/src/client/client_conn.rs

```diff
@@ -340,6 +362,36 @@ pub(super) mod danger {
         pub fn set_certificate_verifier(&mut self, verifier: Arc<dyn ServerCertVerifier>) {
             self.cfg.verifier = verifier;
         }
+
+        /// Create a new dangerous verifier that skips all certificate verification.
+        /// This verifier would only encrypt connection to prevent eavesdropping,
+        /// but will still allow a man-in-the-middle attacks and connections to unverified
+        /// counterparties - i.e. you trust the network not to be compromised.
+        pub fn new_allow_all_verifier() -> Arc<dyn ServerCertVerifier> {
```

Comment:
Conceptually a `new_()` style API doesn't make sense. Like `set_certificate_verifier()`, `DangerousClientConfig` methods should set the internal verifier. So this would be more like `without_verification()`.

Reply:
@djc so something like this?

```rust
// client code
client_config.dangerous().without_verification();

// implementation
    impl<'a> DangerousClientConfig<'a> {
        pub fn without_verification(&mut self) {
            self.set_certificate_verifier(Arc::new(NoCertificateVerification {}))
        }
```

Reply:
Yes, and then including a bunch of docstring that explains the risks and when this could possibly be okay.